src/extensions/base/BaseSchema/BaseSchemaSpecs/index.ts

```diff
@@ -38,11 +40,18 @@ export const BaseSchemaSpecs: ExtensionAuto<BaseSchemaSpecsOptions> = (builder,
         }))
         .addNode(BaseNode.Paragraph, () => ({
             spec: {
+                attrs: {[headingLineNumberAttr]: {default: null}},
                 content: 'inline*',
                 group: 'block',
                 parseDOM: [{tag: 'p'}],
-                toDOM() {
-                    return ['p', 0];
+                toDOM(node) {
+                    const lineNumber = node.attrs[headingLineNumberAttr];
+
+                    return [
+                        'p',
+                        lineNumber === undefined ? {} : {headingLineNumberAttr: lineNumber},
```

Comment:
> `{default: null}`

so maybe check should be `lineNumber === null` ?

> `{headingLineNumberAttr: lineNumber}`

square brackets is missed: `{[headingLineNumberAttr]: lineNumber}`